### PR TITLE
fix: pinterest and algolia cdk test cases

### PR DESCRIPTION
**Pull request by mihir-4116**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/pinterest_tag/transform.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ const commonFieldResponseBuilder = (
   const isValidUserPayload = checkUserPayloadValidity(userPayload);
   if (isValidUserPayload === false) {
     throw new InstrumentationError(
-      "It is required at least one of em, hashed_maids or pair of client_ip_address and client_user_agent"
+      "It is required at least one of em, hashed_maids or pair of client_ip_address and client_user_agent."
```
Comment (saikumarrs) [inline src/v0/destinations/pinterest_tag/transform.js:71]: Revert this change

Comment (mihir-4116) [reply src/v0/destinations/pinterest_tag/transform.js]: done

File: src/v0/destinations/pinterest_tag/transform.js [hunk 2]
```diff
@@ -68,7 +68,7 @@ const commonFieldResponseBuilder = (
   const isValidUserPayload = checkUserPayloadValidity(userPayload);
   if (isValidUserPayload === false) {
     throw new InstrumentationError(
```
Comment (koladilip) [inline src/v0/destinations/pinterest_tag/transform.js:70]: We are using TransformationError in workflows and using InstrumentationError can cause comparison failures. @saikumarrs

File: src/cdk/v2/bindings/default.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-const { TransformationError } = require("../../../v0/util");
+const { TransformationError } = require("../../../v0/util/errorTypes");
```
Comment (koladilip) [inline src/cdk/v2/bindings/default.js:1]: should we use instrumentation error here?

Comment (mihir-4116) [reply src/cdk/v2/bindings/default.js:1]: CC : @saikumarrs

Comment (saikumarrs) [reply src/cdk/v2/bindings/default.js:1]: @koladilip yes. CC @mihir-4116

File: src/v0/destinations/pinterest_tag/transform.js (hunk 2 above)
Comment (saikumarrs) [reply src/v0/destinations/pinterest_tag/transform.js:70]: Yeah, `TransformationError` should be sparingly used. I'll fix this in a separate PR. Let's also deal with the comparison failures separately.

File: src/cdk/v2/bindings/default.js [hunk 4]
```diff
@@ -24,9 +24,9 @@ function isValidEventType(event) {
   return true;
 }
 
-function assert(val, message, status, statTags, destination) {
+function assert(val, message) {
```
Comment (saikumarrs) [inline src/cdk/v2/bindings/default.js:27]: Don't change the signature yet.

**Review by koladilip — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]
